pkg/gatewayserver/io/udp/udp.go

```diff
@@ -32,6 +32,14 @@ import (
 	"go.thethings.network/lorawan-stack/pkg/unique"
 )
 
+// RateLimitingConfig contains configuration settings for the rate limiting
+// capabilities of the UDP gateway frontend firewall.
+type RateLimitingConfig struct {
+	Enable    bool          `name:"enable" description:"Enable rate limiting for gateways"`
+	Messages  int           `name:"messages" description:"Number of past messages to check timestamp for"`
+	Threshold time.Duration `name:"threshold" description:"Filter packet if timestamp is not newer than the older timestamps of the previous messages by this threshold"`
```

Comment:
Not too good at naming new things, these may need to change

Reply:
Name is good